diff -Nur llvm-3.2.src/autoconf/configure.ac llvm-3.2/autoconf/configure.ac
--- llvm-3.2.src/autoconf/configure.ac	2012-11-22 01:13:35.000000000 +0900
+++ llvm-3.2/autoconf/configure.ac	2013-03-20 21:18:46.402839000 +0900
@@ -370,6 +370,7 @@
   hexagon-*)              llvm_cv_target_arch="Hexagon" ;;
   mblaze-*)               llvm_cv_target_arch="MBlaze" ;;
   nvptx-*)                llvm_cv_target_arch="NVPTX" ;;
+  sample-*)               llvm_cv_target_arch="Sample" ;;
   *)                      llvm_cv_target_arch="Unknown" ;;
 esac])
 
@@ -723,6 +724,7 @@
         hexagon)  TARGETS_TO_BUILD="Hexagon $TARGETS_TO_BUILD" ;;
         mblaze)   TARGETS_TO_BUILD="MBlaze $TARGETS_TO_BUILD" ;;
         nvptx)    TARGETS_TO_BUILD="NVPTX $TARGETS_TO_BUILD" ;;
+        sample)   TARGETS_TO_BUILD="Sample $TARGETS_TO_BUILD" ;;
         host) case "$llvm_cv_target_arch" in
             x86)         TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
             x86_64)      TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
diff -Nur llvm-3.2.src/configure llvm-3.2/configure
--- llvm-3.2.src/configure	2012-11-22 01:13:35.000000000 +0900
+++ llvm-3.2/configure	2013-03-20 21:18:46.402839000 +0900
@@ -5419,7 +5419,7 @@
   enableval=host
 fi
 case "$enableval" in
-  all) TARGETS_TO_BUILD="X86 Sparc PowerPC ARM Mips CellSPU XCore MSP430 CppBackend MBlaze NVPTX Hexagon" ;;
+  all) TARGETS_TO_BUILD="X86 Sparc PowerPC ARM Mips CellSPU XCore MSP430 CppBackend MBlaze NVPTX Hexagon Sample" ;;
   *)for a_target in `echo $enableval|sed -e 's/,/ /g' ` ; do
       case "$a_target" in
         x86)      TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
@@ -5438,6 +5438,7 @@
         hexagon)  TARGETS_TO_BUILD="Hexagon $TARGETS_TO_BUILD" ;;
         mblaze)   TARGETS_TO_BUILD="MBlaze $TARGETS_TO_BUILD" ;;
         nvptx)    TARGETS_TO_BUILD="NVPTX $TARGETS_TO_BUILD" ;;
+        sample)   TARGETS_TO_BUILD="Sample $TARGETS_TO_BUILD" ;;
         host) case "$llvm_cv_target_arch" in
             x86)         TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
             x86_64)      TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
diff -Nur llvm-3.2.src/include/llvm/ADT/Triple.h llvm-3.2/include/llvm/ADT/Triple.h
--- llvm-3.2.src/include/llvm/ADT/Triple.h	2012-11-16 06:24:48.000000000 +0900
+++ llvm-3.2/include/llvm/ADT/Triple.h	2013-03-20 21:18:46.402839000 +0900
@@ -67,7 +67,8 @@
     le32,    // le32: generic little-endian 32-bit CPU (PNaCl / Emscripten)
     amdil,   // amdil: amd IL
     spir,    // SPIR: standard portable IR for OpenCL 32-bit version
-    spir64   // SPIR: standard portable IR for OpenCL 64-bit version
+    spir64,  // SPIR: standard portable IR for OpenCL 64-bit version
+    sample   // Sample: sample
   };
   enum VendorType {
     UnknownVendor,
diff -Nur llvm-3.2.src/lib/Support/Triple.cpp llvm-3.2/lib/Support/Triple.cpp
--- llvm-3.2.src/lib/Support/Triple.cpp	2012-11-16 06:24:48.000000000 +0900
+++ llvm-3.2/lib/Support/Triple.cpp	2013-03-20 21:18:46.402839000 +0900
@@ -44,6 +44,7 @@
   case amdil:   return "amdil";
   case spir:    return "spir";
   case spir64:  return "spir64";
+  case sample:  return "sample";
   }
 
   llvm_unreachable("Invalid ArchType!");
@@ -180,6 +181,7 @@
     .Case("amdil", amdil)
     .Case("spir", spir)
     .Case("spir64", spir64)
+    .Case("sample", sample)
     .Default(UnknownArch);
 }
 
@@ -242,6 +244,7 @@
     .Case("amdil", Triple::amdil)
     .Case("spir", Triple::spir)
     .Case("spir64", Triple::spir64)
+    .Case("sample", Triple::sample)
     .Default(Triple::UnknownArch);
 }
 
@@ -678,6 +681,7 @@
   case llvm::Triple::x86:
   case llvm::Triple::xcore:
   case llvm::Triple::spir:
+  case llvm::Triple::sample:
     return 32;
 
   case llvm::Triple::mips64:
@@ -729,6 +733,7 @@
   case Triple::thumb:
   case Triple::x86:
   case Triple::xcore:
+  case Triple::sample:
     // Already 32-bit.
     break;
 
@@ -758,6 +763,7 @@
   case Triple::tce:
   case Triple::thumb:
   case Triple::xcore:
+  case Triple::sample:
     T.setArch(UnknownArch);
     break;
 
diff -Nur llvm-3.2.src/lib/Target/LLVMBuild.txt llvm-3.2/lib/Target/LLVMBuild.txt
--- llvm-3.2.src/lib/Target/LLVMBuild.txt	2012-07-17 03:19:46.000000000 +0900
+++ llvm-3.2/lib/Target/LLVMBuild.txt	2013-03-20 21:18:46.402839000 +0900
@@ -16,7 +16,7 @@
 ;===------------------------------------------------------------------------===;
 
 [common]
-subdirectories = ARM CellSPU CppBackend Hexagon MBlaze MSP430 NVPTX Mips PowerPC Sparc X86 XCore
+subdirectories = ARM CellSPU CppBackend Hexagon MBlaze MSP430 NVPTX Mips PowerPC Sparc X86 XCore Sample
 
 ; This is a special group whose required libraries are extended (by llvm-build)
 ; with the best execution engine (the native JIT, if available, or the
diff -Nur llvm-3.2.src/projects/sample/configure llvm-3.2/projects/sample/configure
--- llvm-3.2.src/projects/sample/configure	2012-10-30 04:49:45.000000000 +0900
+++ llvm-3.2/projects/sample/configure	2013-03-20 21:18:46.402839000 +0900
@@ -5255,7 +5255,7 @@
   enableval=host
 fi
 case "$enableval" in
-  all) TARGETS_TO_BUILD="X86 Sparc PowerPC ARM Mips CellSPU XCore MSP430 Hexagon CppBackend MBlaze NVPTX" ;;
+  all) TARGETS_TO_BUILD="X86 Sparc PowerPC ARM Mips CellSPU XCore MSP430 Hexagon CppBackend MBlaze NVPTX Sample" ;;
   *)for a_target in `echo $enableval|sed -e 's/,/ /g' ` ; do
       case "$a_target" in
         x86)      TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
@@ -5271,6 +5271,7 @@
         cpp)      TARGETS_TO_BUILD="CppBackend $TARGETS_TO_BUILD" ;;
         mblaze)   TARGETS_TO_BUILD="MBlaze $TARGETS_TO_BUILD" ;;
         nvptx)    TARGETS_TO_BUILD="NVPTX $TARGETS_TO_BUILD" ;;
+        sample)   TARGETS_TO_BUILD="Sample $TARGETS_TO_BUILD" ;;
         host) case "$llvm_cv_target_arch" in
             x86)         TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
             x86_64)      TARGETS_TO_BUILD="X86 $TARGETS_TO_BUILD" ;;
